src/main/java/org/testng/internal/TestMethodWorker.java

```diff
@@ -53,13 +55,22 @@ public TestMethodWorker(IInvoker invoker,
                           List<IClassListener> listeners)
   {
     m_invoker = invoker;
-    m_methodInstances = testMethods;
     m_suite = suite;
     m_parameters = parameters;
     m_groupMethods = groupMethods;
     m_classMethodMap = classMethodMap;
     m_testContext = testContext;
     m_listeners = listeners;
+    List<IMethodInstance> localMethodInstances = Lists.newArrayList(testMethods);
```

Comment:
@juherr I fully support the idea of linking to Github issues in the source to explain why this code is here.